### DISCUSSION TITLE
what's the best way to stay up to date on this project (using cached data)

### DISCUSSION THREAD

**Discussion by wavebeem** [PokeAPI/pokeapi#767] [Q&A] [answered]
:wave: hi there! i'm the author of [pkmn.help](https://www.pkmn.help/pokedex/). i wanted to first say massive thanks to y'all for being a fantastic data source for the pokédex on my site.

***what's the best way to stay up to date on this project?***

i do not query the PokéAPI in real time (performance reasons), but i don't want to get severely out of date. with a new pokémon game on the horizon, i want to be able to stay updated on new pokémon additions/updates, without having to do a lot of manual checking all the time.

unfortunately, pulling all the pokeapi data in every CI build is super slow, so i don't want to do that.

i run these two scripts to fetch the pokedex data and download images:

1. https://github.com/wavebeem/pkmn.help/blob/master/bin/scrape-pokeapi.ts
2. https://github.com/wavebeem/pkmn.help/blob/master/bin/fetch-images.ts

**Comment by Naramsim** [comment, marked as answer]
The best for you would be to query our API every time. We are behind Cloudflare so a call is in the order of milliseconds. Whenever we update our APIs, Cloudflare will return the new data. I don't know if this answers your question.

**Reply by wavebeem** [reply]
Perhaps I'm misunderstanding how to use PokéAPI effectively, but I'm currently fetching the list of Pokémon from PokéAPI, then fetching the species, details, and forms for each Pokémon. Resource lists only return links to the API calls to fetch data for each item.

For merely browsing the Pokédex, it would be fine to do a handful of calls, but it still exacerbates situations of extreme client latency by relying on many API calls per page of data.

Also, my website supports offline operation, and I want that to include the Pokédex. It's not too much data to cache inside a web worker, but it is too many API requests to just fire off when the client starts.

There's 3 API requests per Pokémon (species, details, forms), so that's about `900 * 3 = 2700` API calls 

Even if I switched to directly hitting the PokéAPI for browsing in the Pokédex, I also offer an weakness/resistance coverage tool which would require the entire Pokédex to be loaded so I can get type combos for all the Pokémon.

**Reply by C-Garza** [reply]
Maybe you can look at when the pokeapi data gets refreshed? Whenever the data is changed on the `pokeapi` repo, a PR for a data refresh is made in this[ `api-data`](https://github.com/PokeAPI/api-data) repo. Maybe you can run your script updater when these run or selectively choose which ones to run? The PR's in that `api-data` repo have a link to the PR that triggered them.

**Reply by wavebeem** [reply]
Ooh, thanks! I didn't even know about that repo. I'm adding myself as a watcher to the repo for now :) I want to keep my site up to date as soon as y'all get Scarlet/Violet data added to pokeapi. Thanks!

**Reply by Naramsim** [reply]
Whoa, clever solution @C-Garza ! Hahah
